### PR TITLE
Storms from flow-works-http

### PR DESCRIPTION
**Pull request by mjaniakcds**

### CODE REVIEW

File: src/main/scala/carldata/borsuk/storms/ApiObjects.scala [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+      JsObject(
+        "storms" -> JsArray(status.storms.map {
+          _.toJson
+        }.toVector)
+      )
+    }
+
+    def read(value: JsValue): ListStormsResponse = {
+      value.asJsObject().fields("storms") match {
+        case JsArray(arr) => {
```
Comment (wenta) [inline src/main/scala/carldata/borsuk/storms/ApiObjects.scala]: Redudant braces

File: src/main/scala/carldata/borsuk/storms/ApiObjects.scala [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+        "start-date" -> JsString(response.startDate.toString),
+        "end-date" -> JsString(response.endDate.toString),
+        "maxIntensity" -> JsNumber(response.maxIntensity))
+    }
+
+    def read(value: JsValue): GetStormsResponse = {
+      val fields = value.asJsObject.fields
+      val startDate: LocalDateTime = DateTimeHelper.dateParse(fields("start-date").toString)
+      val endDate: LocalDateTime = DateTimeHelper.dateParse(fields("start-date").toString)
+      val maxIntensity = fields.get("maxIntensity").map(doubleFromValue).getOrElse(0.0)
```
Comment (wenta) [inline src/main/scala/carldata/borsuk/storms/ApiObjects.scala]: I think we should `Double.Nan` instead `0.0`. `0.0` suggest later that there was a value.

File: src/main/scala/carldata/borsuk/storms/ApiObjects.scala [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+        "id" -> storm.id.toJson,
+        "start-date" -> storm.startDate.toString.toJson,
+        "end-date" -> storm.endDate.toString.toJson
+      )
+    }
+
+    def read(json: JsValue): ApiObjects.StormsObject = {
+      val fields = json.asJsObject.fields
+      val id: String = fields("id").toString
+      val startDate: LocalDateTime = DateTimeHelper.dateParse(fields("startDate").toString)
```
Comment (wenta) [inline src/main/scala/carldata/borsuk/storms/ApiObjects.scala]: In read we have `startDate` & `endDate`, while in write `start-date` and `end-date`. Names should be the same.

File: src/main/scala/carldata/borsuk/storms/ApiObjects.scala [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+    }
+  }
+
+  /**
+    * FitParams formatter
+    */
+  implicit object FitStormsParamsFormat extends RootJsonFormat[FitStormsParams] {
+    def write(params: FitStormsParams): JsObject = {
+      JsObject(
+        "start-date" -> JsString(params.startDate.toString),
```
Comment (wenta) [inline src/main/scala/carldata/borsuk/storms/ApiObjects.scala]: What about `resolution`, `windowSize` & `intensitySize` ?

File: src/main/scala/carldata/borsuk/storms/Storms.scala [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+    if (xs.nonEmpty) xs.dataPoints.maxBy(_._2)._2
+    else 0.0
+  }
+
+  /**
+    * List all storms
+    */
+  def list(): Seq[(String, Session)] = {
+    model.map(x =>
```
Comment (wenta) [inline src/main/scala/carldata/borsuk/storms/Storms.scala]: It is so small, lets make it one-line

File: src/main/scala/carldata/borsuk/storms/Storms.scala [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    * Find all storms sessions with intensity
+    */
+  def find(rainfall: TimeSeries[Double], windowSize: Duration, intensitySize: Duration): Seq[(String, Session, Double)] = {
+    if (rainfall.isEmpty) Seq()
+    else {
+      val rainSessions = Sessions.findSessions(rainfall, windowSize)
+
+      rainSessions
+        .map(x => (randomUUID().toString, x, maxIntensity(x, rainfall, intensitySize)))
+        .filter(x => x._3 > 0)
```
Comment (wenta) [inline src/main/scala/carldata/borsuk/storms/Storms.scala]: Could we addtiotionaly sort it by intesity?

File: src/main/scala/carldata/borsuk/storms/ApiObjects.scala (hunk 1 above)
Comment (mjaniakcds) [reply src/main/scala/carldata/borsuk/storms/ApiObjects.scala]: done

File: src/main/scala/carldata/borsuk/storms/ApiObjects.scala (hunk 2 above)
Comment (mjaniakcds) [reply src/main/scala/carldata/borsuk/storms/ApiObjects.scala]: corrected

File: src/main/scala/carldata/borsuk/storms/ApiObjects.scala (hunk 3 above)
Comment (mjaniakcds) [reply src/main/scala/carldata/borsuk/storms/ApiObjects.scala]: changed

File: src/main/scala/carldata/borsuk/storms/ApiObjects.scala (hunk 4 above)
Comment (mjaniakcds) [reply src/main/scala/carldata/borsuk/storms/ApiObjects.scala]: added

File: src/main/scala/carldata/borsuk/storms/Storms.scala (hunk 5 above)
Comment (mjaniakcds) [reply src/main/scala/carldata/borsuk/storms/Storms.scala]: One line indeed

File: src/main/scala/carldata/borsuk/storms/Storms.scala (hunk 6 above)
Comment (mjaniakcds) [reply src/main/scala/carldata/borsuk/storms/Storms.scala]: Yes we could

**Pull request merged**